lib/transaction.js

```diff
@@ -49,53 +49,7 @@ class Transaction extends EventEmitter {
     //      2 separate Promise Chains.  This ensures that each Promise Chain
     //      can establish its error-handling semantics without interfering
     //      with the other Promise Chain.
-    const basePromise = this.acquireConnection(config, (connection) => {
-      const trxClient = (this.trxClient = makeTxClient(
-        this,
-        client,
-        connection
-      ));
-      const init = client.transacting
-        ? this.savepoint(connection)
-        : this.begin(connection);
-      const executionPromise = new Promise((resolver, rejecter) => {
-        this._resolver = resolver;
-        this._rejecter = rejecter;
-      });
-
-      init
-        .then(() => {
-          return makeTransactor(this, connection, trxClient);
-        })
-        .then((transactor) => {
-          transactor.executionPromise = executionPromise;
-
-          // If we've returned a "thenable" from the transaction container, assume
-          // the rollback and commit are chained to this object's success / failure.
-          // Directly thrown errors are treated as automatic rollbacks.
-          let result;
-          try {
-            result = container(transactor);
-          } catch (err) {
-            result = Promise.reject(err);
-          }
-          if (result && result.then && typeof result.then === 'function') {
-            result
-              .then((val) => {
-                return transactor.commit(val);
-              })
-              .catch((err) => {
-                return transactor.rollback(err);
-              });
-          }
-          return null;
-        })
-        .catch((e) => {
-          return this._rejecter(e);
-        });
-
-      return executionPromise;
-    });
+    const basePromise = this._kickThingsOff(config, container);
```

Comment:
`_kickThingsOff` this sounds slightly too informal and vague, can we make it more specific?

Reply:
Yeah -- I haven't been able to think of a good name for it so far.  My guess is that this function will most likely disappear by the time the overall refactoring has been completed.

It's a bit tricky to pick a name because this function is currently doing too many things.  Ex:

1. Ensuring that the previous Transaction has completed
2. Acquiring a connection
3. (Possibly) setting up disposal for the connection
4. Creating a Transactor
5. (Possibly) setting up commit/rollback guarantees for the Transactor
6. Yielding control back to the caller

So yeah... it's tricky to come up w/ a name that encompasses all of those things.  lol

Reply:
How about: `_evaluateContainer(..)` ?